### ISSUE TITLE
Cannot `linkcheck`  url with unicode.

### ISSUE THREAD

**Issue by Carreau** [sphinx-doc/sphinx#2171]
example url:

```
https://github.com/search?utf8=✓&q=is%3Aissue+is%3Aopen+is%3Asprint-friendly+user%3Ajupyter&type=Issues&ref=searchresults
```

offending part is `utf8=✓` I guess. 

Fails [here](https://github.com/sphinx-doc/sphinx/blob/6f6aa53f372c25ec781f0e2c26fa7433f03342cc/sphinx/builders/linkcheck.py#L157-L159):

```
req_url = (split[0].encode() + '://' +       # scheme
           split[1].encode('idna') +         # netloc
           quote(split[2].encode('utf-8')))  # path
```

As this is concatenate respectively  bytes, string, bytes, string.

**Comment by tk0miya** [comment]
Fixed at 9669f2a

**Comment by Carreau** [comment]
Awesome thanks for the quick fix !
